source/client-side-operations-timeout/tests/runCursorCommand.yml

```
@@ -0,0 +1,316 @@
+description: runCursorCommand
+
+schemaVersion: '1.9'
+
+runOnRequirements:
+  - minServerVersion: "4.4"
+
+createEntities:
+  - client:
+      id: &failPointClient failPointClient
+      useMultipleMongoses: false
+  - client:
+      id: &commandClient commandClient
+      useMultipleMongoses: false
+      observeEvents: [commandStartedEvent, commandSucceededEvent]
+  - client:
+      id: &client client
+      useMultipleMongoses: false
+      observeEvents: [commandStartedEvent]
+  - database: # For tests that need success event assertions
+      id: &commandDb commandDb
+      client: *commandClient
+      databaseName: *commandDb
+  - database:
+      id: &db db
+      client: *client
+      databaseName: *db
+  - collection:
+      id: &collection collection
+      database: *db
+      collectionName: *collection
+
+initialData:
+  - collectionName: *collection
+    databaseName: *db
+    documents: &documents
+      - { _id: 1, x: 11 }
+      - { _id: 2, x: 22 }
+      - { _id: 3, x: 33 }
+      - { _id: 4, x: 44 }
+      - { _id: 5, x: 55 }
+
+tests:
+  - description: errors if timeoutMode is set without timeoutMS
+    operations:
+      - name: runCursorCommand
+        object: *db
+        arguments:
+          commandName: find
+          command: { find: *collection }
+          timeoutMode: cursorLifetime
+        expectError:
+          isClientError: true
+
+  - description: error if timeoutMode is cursorLifetime and cursorType is tailableAwait
+    operations:
+      - name: runCursorCommand
+        object: *db
+        arguments:
+          commandName: find
+          command: { find: *collection }
+          timeoutMode: cursorLifetime
+          cursorType: tailableAwait
+        expectError:
+          isClientError: true
+
+  - description: error if getMore maxTimeMS setting is greater than timeoutMS
+    operations:
+      - name: runCursorCommand
+        object: *collection
+        arguments:
+          commandName: find
+          command: { find: *collection }
+          timeoutMS: 5
+          maxTimeMS: 10
+        expectError:
+          isClientError: true
+
+  - description: error if getMore maxTimeMS setting is equal to timeoutMS
+    operations:
+      - name: runCursorCommand
+        object: *collection
```

```suggestion
        object: *db
```

runCursorCommand is [defined](https://github.com/mongodb/specifications/pull/1412/files#diff-f718c246f51e05b3731a7b03b176a0d4b0c381a9261810742d2677823f963f98R1806) as an operation on the database object, not the collection object.

---

fixed